packages/select/src/withDelete.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  getAboveNode,
+  isBlockAboveEmpty,
+  isSelectionExpanded,
+  PlateEditor,
+  queryNode,
+  removeNodes,
+  Value,
+  WithPlatePlugin,
+} from '@udecode/plate-common';
+
+import { DeletePlugin } from './createDeletePlugin';
+
+/**
+ * Set a list of element types to select on backspace
+ */
+export const withDelete = <
+  V extends Value = Value,
+  E extends PlateEditor<V> = PlateEditor<V>,
+>(
+  editor: E,
+  { options: { query } }: WithPlatePlugin<DeletePlugin, V, E>
+) => {
+  const { deleteForward } = editor;
+  editor.deleteForward = (unit) => {
+    if (!editor.selection) return;
+    const isValidNode = queryNode(getAboveNode(editor), query);
```

Comment:
```suggestion
    const isValidNode = !query || queryNode(getAboveNode(editor), query);
```

Reply:
I don't think this is necessary

Reply:
`queryNode` returns true if query is undefined, I think?

Reply:
This prevents computing `getAboveNode(editor)` when not defined

Reply:
done